blank-canvas-blocks/sass/blocks/_button.scss

```diff
@@ -4,36 +4,72 @@
 
 // NOTE: These remain for the styling of buttons that are NOT blocks and is used elsewhere.  This can be removed when those no longer exist.
 @mixin button-main-styles {
- 	font-weight: var(--wp--custom--button--typography--font-weight);
-	font-family: var(--wp--custom--button--typography--font-family);
-	font-size: var(--wp--custom--button--typography--font-size);
-	line-height: var(--wp--custom--button--typography--line-height);
-	text-decoration: none; // Needed because link styles inside .entry-content add a text decoration
+	@include button-padding-styles;
+	@include button-typography-styles;
+	@include button-color-styles;
+	border-radius: var(--wp--custom--button--border--radius);
+}
+
+@mixin button-color-styles {
 	color: var(--wp--custom--button--color--text);
 	background-color: var(--wp--custom--button--color--background);
-	padding: calc(.667em + 2px) calc(1.333em + 2px); //The padding found on an unmodified Button Block
 	svg {
 		fill: var(--wp--custom--button--color--text);
 	}
 }
 
+@mixin button-padding-styles {
+	padding-top: calc( var(--wp--custom--button--spacing--padding--top) + var(--wp--custom--button--border--width) );
+	padding-bottom: calc( var(--wp--custom--button--spacing--padding--bottom) + var(--wp--custom--button--border--width) );
+	padding-left: calc( var(--wp--custom--button--spacing--padding--left) + var(--wp--custom--button--border--width) );
+	padding-right: calc( var(--wp--custom--button--spacing--padding--right) + var(--wp--custom--button--border--width) );
+}
+
+@mixin button-typography-styles {
+ 	font-weight: var(--wp--custom--button--typography--font-weight);
+	font-family: var(--wp--custom--button--typography--font-family);
+	font-size: var(--wp--custom--button--typography--font-size);
+	line-height: var(--wp--custom--button--typography--line-height);
+	text-decoration: none; // Needed because link styles inside .entry-content add a text decoration
+}
+
+@mixin button-border-styles {
+	border-style: var(--wp--custom--button--border--style);
+	border-color: currentColor;
```

Comment:
Should this be `currentColor` or `var(--wp--custom--button--border--color)`?

Reply:
currentColor causes the TEXT and OUTLINE to match when in border styles. i.e. changing the "text color" of a button also changed the outline color.
I'm not sure what the standard or expected behavior was there; trunk wasn't behaving at all to compare against, but that was the behavior I was expecting (relatively naively).

Reply:
I think it should be currentColor, since that is what the editor is doing too

Reply:
Setting it to 'currentColor' allows the border to be the same color as the label, which gives the user the ability to change both by changing the 'color' property.  I tried to make that a theme decision (setting the value of --wp--custom--button--border--color to `currentColor`) but then there is no mechanism by which to actually choose a color it should default to.  Doing it this way locks in the opinion that changing the color changes the border color too.  I'm open to suggestions refactoring it, but I think I've tried all the ways I could think of and there were blockers of some kind or another all the other ways I tried to make it behave.